### ISSUE TITLE
flutter

### ISSUE THREAD

**Issue by Solido** [Dana-Ferguson/time_machine#2]
Message with the last Flutter upgrate :

```
Because every version of flutter_test from sdk depends on collection 1.14.6
and time_machine >=0.8.0 depends on collection ^1.14.10, flutter_test from
sdk is incompatible with time_machine >=0.8.0.

So, because flutter_calender depends on both time_machine 0.8.2 
and flutter_test any from sdk, version solving failed.
```

**Comment by Dana-Ferguson** [comment]
Thank you for informing me of this. I will get working on it tonight.

**Comment by Solido** [comment]
Thank you, I really wanted to try your lib.
I Feel lucky for tomorrow !

**Comment by Dana-Ferguson** [comment]
**TLDR**: You're good to go. Let me know how it goes. We'll close this issue when we know you're good.

I went ahead and tried to run my unit tests with `flutter test`. The non-dev dependencies should be fine now (looks like `collection` was the only one). Unfortunately, because of an issue with `TypeMatcher` in `package:matcher/matcher.dart` changing from an abstract class (in the version used by Flutter) to an instanced class (in the version used by time_machine), I can't direct unit test them -- well, I technically can, but the tests fail, since `testing/time_matchers.dart` is an import into time_machine's testing framework (which includes a usage of `TypeMatcher`).

Because of:
 * 'The flutter test command lets you run your tests in a local Dart VM with a headless version of the Flutter Engine, which supplies these libraries.' (from [here](https://flutter.io/testing/))
 * flutter not being a dependency of time_machine

I didn't \ don't (??? I'm unsure about this) believe that running `flutter test` yields any extra benefit over regular `pub run test`. I think after `TypeMatcher` gets updated in [flutter_test/pubspec.yaml](https://github.com/flutter/flutter/blob/master/packages/flutter_test/pubspecz.yaml) and pushed live, running flutter tests directly may not be all that hard.

I've pushed 0.8.3. This should allow you to run unit tests that depend on time_machine in Flutter. If something goes wrong, get back to me.

**Comment by Solido** [comment]
Some new tentatives and their results :
` String tz = await FlutterNativeTimezone.getLocalTimezone();
    await TimeMachine.initialize({'rootBundle': rootBundle, 'timeZone': tz});`
```
flutter: NoSuchMethodError: Class '_InternalLinkedHashMap<dynamic, dynamic>' has no instance method 'load'.
Receiver: _LinkedHashMap len:2
Tried calling: load("packages/time_machine/data/tzdb/tzdb.bin")
flutter: #0      Object.noSuchMethod (dart:core/runtime/libobject_patch.dart:46:5)
#1      _FlutterMachineIO.getBinary (package:time_machine/src/platforms/vm.dart:44:38)
<asynchronous suspension>
#2      TzdbIndex.loadAll (package:time_machine/src/timezones/tzdb_io.dart:35:41)
<asynchronous suspension>
#3      TzdbDateTimeZoneSource._init (package:time_machine/src/timezones/tzdb_datetimezone_source.dart:49:42)
<asynchronous suspension>
#4      TzdbDateTimeZoneSource._tzdbIndexAsync (package:time_machine/src/timezones/tzdb_datetimezone_source.dart:68:9)
#5      TzdbDateTimeZoneSource._tzdbIndexAsync (package:time_machine/src/timezones/tzdb_datetimezone_source.dart:66:28)
#6      TzdbDateTimeZoneSource.getIds (package:time_machine/src/timezones/tzdb_datetimezone_source.dart:77:53)
<asynchronous suspension>
#7      DateTimeZoneCache.getCache (package:time_machine/src/timezones/datetimezone_cache.dart:54:36)
<asynchronous suspension>
#8      DateTimeZoneProviders.tzdb (package:time_machine/src/time<…>

```

**Comment by Solido** [comment]
Copy Paste from the instructions : timezone is undefined :)
`await TimeMachine.initialize({rootBundle: rootBundle, timeZone: tz});`

**Comment by Solido** [comment]
Maybe you could have your own Flutter source under example ?

**Comment by Dana-Ferguson** [comment]
Yeah. I have one ok my local machine ( how I've been testing ). You only
include the timeZone part, if you're also using the package
https://pub.dartlang.org/packages/flutter_native_timezone.

**Comment by Dana-Ferguson** [comment]
https://github.com/Dana-Ferguson/time_machine/tree/master/example/flutter_example

It's not great, certaintly not pretty,, but I uploaded the yaml + lib/main.dart from my Flutter project. (Sorry for the weird'ish-ness, It's just approaching 6 am, and I was up late).

**Comment by Dana-Ferguson** [comment]
![image](https://user-images.githubusercontent.com/7284858/43044556-f4d055f2-8d75-11e8-888b-e2cfffef9b93.png)

Okay, it's working. I need to, do another push for 0.8.4.

**Comment by Solido** [comment]
Success ! Thank you very much !
Will work on this on the next hours.

**Comment by Dana-Ferguson** [comment]
Issue was: https://github.com/Dana-Ferguson/time_machine/commit/736e1fe7358879d3e3978672aad66374c636fd3f#diff-2d7be75422a1489dc4a9d50333bab61c

Sorry about that. Ping me again if you're still having  issues.

**Comment by Solido** [comment]
Will ping you after you get some rest ;)

**Comment by Solido** [comment]
Thank you so much !
All is working well ! very helpful package.
